src/vs/base/browser/markdownRenderer.ts

```diff
@@ -15,12 +15,15 @@ import { cloneAndChange } from 'vs/base/common/objects';
 import { escape } from 'vs/base/common/strings';
 import { URI } from 'vs/base/common/uri';
 import { Schemas } from 'vs/base/common/network';
+import { renderCodicons } from 'vs/base/browser/ui/codiconLabel/codiconLabel';
 
 export interface MarkdownRenderOptions extends FormattedTextRenderOptions {
 	codeBlockRenderer?: (modeId: string, value: string) => Promise<string>;
 	codeBlockRenderCallback?: () => void;
 }
 
+const codiconsRegex = /^codicon:\/\/(.*)$/;
```

Comment:
Hm, I wonder if the actual codicon should be the path, not the auth part, e.g. `codicon:zap` or both, or more hierarchical via `icon://codeicon/zap`?

Reply:
the `icon://auth/name` part would have the advantage that it's open for contributions, e.g we could expose other contributed icons like that, for instance `icon://foo.extension/bar-name

Reply:
OK, that sounds good. I'll make that change.

Reply:
FYI - I will also invest some time to see what it takes to support this custom trees and (stretch) in menu items

Reply:
It is now using `icon://vscode.codicons/git-commit` url format now. Should allow for different builtin icon styles as well as can be extended to extensions using `icon://foo.extension/icon-name` as you suggested.